### PR TITLE
Feature/group endpoints

### PR DESCRIPTION
**Pull request by peavers**
Adds functions to fetch information about a users groups

### CODE REVIEW

File: src/main/scala/com/codacy/client/stash/client/StashClient.scala [hunk 1]
```diff
@@ -191,6 +191,8 @@ class StashClient(apiUrl: String, authenticator: Option[Authenticator] = None) {
       .getOrElse(Right(json))
   }
 
-  private def generateUrl(endpoint: String) = s"$apiUrl$endpoint"
+  private def generateUrl(endpoint: String) = {
+    s"$apiUrl${endpoint.replace(" ", "-")}"
```
Comment (machadoit) [inline src/main/scala/com/codacy/client/stash/client/StashClient.scala]: What sorcery is this? :D

File: src/main/scala/com/codacy/client/stash/service/AdminServices.scala [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    * The authenticated user must have the LICENSED_USER permission to call this resource.
+    */
+  def findUsersInGroup(
+      context: String,
+      filter: Option[String],
+      pageRequest: Option[PageRequest]
+  ): RequestResponse[Seq[User]] = pageRequest match {
+    case Some(pageRequest) =>
+      client.executePaginated(
+        Request(s"$BASE/groups/more-members?context=$context&filter=${filter.getOrElse("")}", classOf[Seq[User]]),
```
Comment (machadoit) [inline src/main/scala/com/codacy/client/stash/service/AdminServices.scala]: I would prefer to don't write "filter=" even if this work. The "&filter = filterValue" should be all the "filterField", and that should wither have the content or be empty string

File: src/main/scala/com/codacy/client/stash/service/RepositoryServices.scala [hunk 3]
```diff
@@ -74,6 +74,22 @@ class RepositoryServices(client: StashClient) {
...
+      pageRequest: Option[PageRequest]
+  ): RequestResponse[Seq[Group]] = pageRequest match {
+    case Some(pageRequest) =>
+      client.executePaginated(
+        Request(s"$BASE/$projectKey/repos/$repositorySlug/permissions/groups", classOf[Seq[Group]]),
+        pageRequest
+      )
+    case None =>
+      client.executePaginated(
+        Request(s"$BASE/$projectKey/repos/$repositorySlug/permissions/groups", classOf[Seq[Group]])
```
Comment (machadoit) [inline src/main/scala/com/codacy/client/stash/service/RepositoryServices.scala]: It seems a bit of a shame that we repeat the "Request string" every timen it seems that the executePaginated method should receive an option

Comment (peavers) [reply src/main/scala/com/codacy/client/stash/service/RepositoryServices.scala]: Its overloaded as to not break all the other implementations throughout the code. At this point in time I'm alright with it being like this.

File: src/main/scala/com/codacy/client/stash/client/StashClient.scala (hunk 1 above)
Comment (peavers) [reply src/main/scala/com/codacy/client/stash/client/StashClient.scala]: Quick magic to deal with the fact that Bitbucket allows spaces in the slugs when creating repositories and returns it that way to us via their API, however it doesn't accept the same URL back in, it requires the space to be replaced with a `-`. Quick wins; if you've got another quick suggestion however?

Comment (peavers) [reply src/main/scala/com/codacy/client/stash/client/StashClient.scala]: Magic! It actually isn't supposed to be in this PR!

File: src/main/scala/com/codacy/client/stash/service/AdminServices.scala [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      filter: Option[String],
+      pageRequest: Option[PageRequest]
+  ): RequestResponse[Seq[User]] = {
+    val parameters = Map("context" -> context, "filter" -> filter.getOrElse(""))
+    val filterField = filter.map(value => s"filter=$value").getOrElse("")
+
+    pageRequest match {
+      case Some(pageRequest) =>
+        client.executePaginatedWithPageRequest(
+          Request(s"$BASE/groups/more-members?context=$context&$filterField", classOf[Seq[User]]),
```
Comment (machadoit) [inline src/main/scala/com/codacy/client/stash/service/AdminServices.scala]: & to be part of the filterField plz

**Pull request merged**